src/ui/public/index_patterns/_get.js

```diff
@@ -1,38 +1,40 @@
 import _ from 'lodash';
 import { SavedObjectsClientProvider } from 'ui/saved_objects';
 
-export function IndexPatternsGetIdsProvider(Private) {
+export function IndexPatternsGetProvider(Private) {
   const savedObjectsClient = Private(SavedObjectsClientProvider);
 
   // many places may require the id list, so we will cache it separately
   // didn't incorporate with the indexPattern cache to prevent id collisions.
-  let cachedPromise;
+  const cachedPromises = {};
 
-  const getIds = function () {
-    if (cachedPromise) {
+  const get = function (field) {
+    if (cachedPromises[field]) {
       // return a clone of the cached response
-      return cachedPromise.then(function (cachedResp) {
+      return cachedPromises[field].then(function (cachedResp) {
         return _.clone(cachedResp);
       });
     }
 
-    cachedPromise = savedObjectsClient.find({
+    cachedPromises[field] = savedObjectsClient.find({
       type: 'index-pattern',
       fields: [],
       perPage: 10000
     }).then(resp => {
-      return resp.savedObjects.map(obj => obj.id);
+      return resp.savedObjects.map(obj => _.get(obj, field));
     });
 
     // ensure that the response stays pristine by cloning it here too
-    return cachedPromise.then(function (resp) {
+    return cachedPromises[field].then(function (resp) {
       return _.clone(resp);
     });
   };
 
-  getIds.clearCache = function () {
-    cachedPromise = null;
+  return (field) => {
+    const getter = get.bind(get, field);
+    getter.clearCache = function () {
```

Comment:
Previously the `IndexPatternsProvider` would clear the complete cache whenever it was invalidated. With this change, only the part of the cache used by the respective user is cleared. This seems like a dangerous way to end up with everything sitting in memory forever.

I wonder if we should _only_ cache IDs, and simply fetch other things on demand?